### ISSUE TITLE
Automatically create badges when competition finish

### ISSUE THREAD

**Issue by skyksandr** [skyderby/skyderby#122]
## Current behavior

When competition finish organizers ask me to create badges for the winners.

## How to do it better 

When competition finish automatically create badges for winners in every category

## How to make it

1. Add polymorphic association to badges `source`
2. Add scoped controllers `event_finalizations`, `tournament_finalizations` with one action `create`. When invoked it should:
* delete existing badges 
* find winners for event and create badges for them

**Comment by skyksandr** [comment]
@MariiaAlehina, can you take a look on this?

**Comment by MariiaAlehina** [comment]
@skyksandr Hello. Yes. I am on it. But as I see there is connection Badge -> Profile. Do I have to change type to `polymorphic `for it?

**Comment by skyksandr** [comment]
That's right -`badge` `belongs_to` `profile`, and it should also belongs to event where it achieved.

**Comment by MariiaAlehina** [comment]
ok. Thanks.

**Comment by MariiaAlehina** [comment]
@skyksandr Can you please explain how do I can collect all winers. As I see there are several _kinds_ of events.
Simple Event and Event in Tournament. Tournament has slots and there is `winner_id` field  only there.
Thanks.
